src/marvin/utilities/python.py

```diff
@@ -0,0 +1,95 @@
+import inspect
+import textwrap
+from typing import Any, Callable, List, Optional
+
+from pydantic import BaseModel, Field, computed_field
+
+from marvin.utilities.asyncio import run_sync
+
+
+class ParameterModel(BaseModel):
+    name: str
+    annotation: Optional[str]
+    default: Optional[str]
+
+
+class PythonFunction(BaseModel, arbitrary_types_allowed=True):
+    function: Callable = Field(description="Original function object")
```

Comment:
could we use `model_config` instead?
```suggestion
from pydantic import ConfigDict
...
class PythonFunction(BaseModel):
    model_config = ConfigDict(arbitrary_types_allowed=True)
    function: Callable = Field(description="Original function object")
```